priv/emqx.schema

```diff
@@ -206,6 +206,21 @@ end}.
   {override_env, "NODE_NAME"}
 ]}.
 
+%% Node name regular expression from net_kernel:valid_name_head/1.
+%% Be same with otp.
+{translation, "vm_args.-name",
+fun(Conf) ->
+  NodeName = cuttlefish:conf_get("node.name", Conf),
+  {Head, _Host} = lists:splitwith(fun(C) -> C =/= $@ end, NodeName),
+  {ok, MP} = re:compile("^[0-9A-Za-z_\\-]+$", [unicode]),
+  case re:run(Head, MP) of
```

Comment:
do the check in `bin/emqx` as the name can be overridden from environment variables.

Reply:
I tried this:
```shell
EMQX_NODE_NAME=emqx.1@127.0.0.1 ./_build/emqx/rel/emqx/bin/emqx console
2022-05-18T14:31:41.926373+08:00 [error] Error generating configuration in phase apply_translations
2022-05-18T14:31:41.926487+08:00 [error] Translation for 'vm_args.-name' found invalid configuration: Bad node name, expect ^[0-9A-Za-z_\-]+$
log.to = "console"
node.name = "emqx.1@127.0.0.1"
```
Seems to be working fine.